sha.go

```diff
@@ -63,11 +63,17 @@ func SHA512(p []byte) (sum [64]byte) {
 	return
 }
 
-// Same as SupportsHKDF, as in v1.1.1+
-func SupportsSHA3() bool {
-	return vMajor > 1 ||
-		(vMajor >= 1 && vMinor > 1) ||
-		(vMajor >= 1 && vMinor >= 1 && vPatch >= 1)
+// SupportsHash returns true if h is supported.
```

Comment:
Given the comment above about RSA and MD5, this seems too general and I wonder what "supported" means. Just a little extra detail seems like enough to me:

```suggestion
// SupportsHash returns true if a hash.Hash implementation is supported for h.
```